.github/workflows/build.yml

```diff
@@ -189,7 +195,62 @@ jobs:
         SSH_USER: mixxx
         UPLOAD_ID: ${{ github.run_id }}
 
+    - run: |
+        if ! gh release view --repo "${{ github.repository }}" "${RELEASE_NAME}" &> /dev/null
+        then 
+          gh release create --repo "${{ github.repository }}" "${RELEASE_NAME}" \
+            --generate-notes \
+            --latest \
+            --target "${{ github.push.after }}" ${{ !endsWith( github.event.ref, '-rel' ) && '--prerelease' || '' }}
+        fi
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        RELEASE_NAME: ${{ env.MIXXX_VERSION }}-${{ steps.vars.outputs.sha_short }}
+      if: ${{ github.event_name == 'push' }}
+      id: ghrelease
+      shell: bash
+      name: "Creates a release in GitHub if it doesn't exist yet"
+
+    - run: |
+        vcpkgFileName="${DEPS_BASE_NAME}-${MIXXX_VERSION}-${VCPKG_TRIPLET}-${SHA_SHORT}"
+        sha256sum "${VCPKG_ASSET_PATH}" | cut -d' ' -f 1 > "${vcpkgFileName}.zip.sha256"
+        split -b1536m -d "${VCPKG_ASSET_PATH}" "${vcpkgFileName}.zip.part"
+        gh release upload --repo "${{ github.repository }}" "${RELEASE_NAME}" ${vcpkgFileName}.zip*
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        SHA_SHORT: ${{ steps.vars.outputs.sha_short }}
+        VCPKG_TRIPLET: ${{ matrix.vcpkg_triplet }}
+        RELEASE_NAME: ${{ env.MIXXX_VERSION }}-${{ steps.vars.outputs.sha_short }}
+        VCPKG_ASSET_PATH: ${{ matrix.vcpkg_path }}/${{ env.DEPS_BASE_NAME }}-${{ env.MIXXX_VERSION }}-${{ matrix.vcpkg_triplet }}-${{ steps.vars.outputs.sha_short }}.zip
+      shell: bash
+      if: runner.os != 'Windows' && steps.ghrelease.outcome == 'success'
+      name: "[MacOS] Creates assets and upload to GitHub release"
+
+    - run: |
+        $stream = [System.IO.File]::OpenRead("${{ matrix.vcpkg_path }}/${{ env.VCPKG_ASSET_PATH }}")
+        $chunkNum = 0
+        $bufSize = 1536MB
+        $barr = New-Object byte[] $bufSize
+
+        while( $bytesRead = $stream.Read($barr,0,$bufSize)){
+          $outFile = "${{ env.VCPKG_ASSET_PATH }}.part$chunkNum"
+          $ostream = [System.IO.File]::OpenWrite($outFile)
+          $ostream.Write($barr,0,$bytesRead);
+          $ostream.close();
+          echo "wrote $outFile"
+          $chunkNum += 1
+        }
+        (Get-FileHash -Algorithm SHA256 -Path "${{ matrix.vcpkg_path }}/${{ env.VCPKG_ASSET_PATH }}").Hash | Out-File -FilePath "${{ env.VCPKG_ASSET_PATH }}.sha256"
```

Comment:
It would be nice to have the checksum as action artefact as well. You may also consider to remove the extra checksum generation from our deploy script, to have only a single place where it is done. Remember that it reads GBs.

Reply:
> It would be nice to have the checksum as action artefact as well

Do you mean to generate the checksum on the flight, when uploading? While being [a requested feature](https://github.com/cli/cli/issues/1711), it doesn't seem to be possible currently.

> You may also consider to remove the extra checksum generation from our deploy script, to have only a single place where it is done.

Not sure what you mean - are you saying we shouldn't run the checksum on our `buildenv` script in the Mixxx repo? 

> Remember that it reads GBs.

Yes, but from a local SSD harddrive. This resource consumption can likely be negligible as we are probably talking <10Wh worth of induced current/month. Or did you mean read from somewhere else?

Reply:
No, it would be just a performance benefit to do the checksum calculation only once. Currently it is done in de deploy script and in the workflow. 

It is a minor annoyance, that the windows sh256 file has a different format. 

I have no experience what is most convenient for a manual check on the users systems though.

Reply:
> it is done in de deploy script

Are you referring to `vcpkg export`? 

> and in the workflow

VCPKG workflow or Mixxx?